File: projekatN/src/main/java/com/example/model/Apartment.java
package com.example.model;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity
public class Apartment {
	@Id
	@GeneratedValue
	private Long id;

	private Integer number;
	private String description;
	@ManyToOne(fetch = FetchType.EAGER)
	private Building building;

	public Apartment() {
	}

	public Apartment(Long id, Integer number, String description) {
		super();
		this.id = id;
		this.number = number;
		this.description = description;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Integer getNumber() {
		return number;
	}

	public void setNumber(Integer number) {
		this.number = number;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public Building getBuilding() {
		return building;
	}

	public void setBuilding(Building building) {
		this.building = building;
	}

}

File: frontend/src/main/frontend/src/app/models/apartment.ts
import { User } from "./user";
import { Building } from "./building";

export class Apartment implements ApartmentInterface{
	public id: number;
	public description: string;
    public number: number;
	public owner: User;
	public building: Building;
		
	constructor(apartemntCfg:ApartmentInterface)
	{	
		this.id = apartemntCfg.id;
		this.description = apartemntCfg.description;
        this.number = apartemntCfg.number;
		this.owner = apartemntCfg.owner;
		this.building = apartemntCfg.building;
	}
}

interface ApartmentInterface{
	id?: number;
    description: string;
    number: number;
	owner: User;
	building?: Building;
}